splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/UpgradeSystemProcedures.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2012 - 2020 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.derby.impl.sql.catalog.upgrade;
+
+import com.splicemachine.EngineDriver;
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.procedures.ProcedureUtils;
+
+import java.sql.*;
+import java.util.List;
+
+public class UpgradeSystemProcedures {
+
+    public static void BEGIN_ROLLING_UPGRADE(final ResultSet[] resultSets) throws SQLException, StandardException {
+        try {
+            UpgradeManager upgradeManager = EngineDriver.driver().manager().getUpgradeManager();
+            upgradeManager.startRollingUpgrade();
+            resultSets[0] = ProcedureUtils.generateResult("Success", "The system is rolling upgrade.");
+        }
+        catch (Exception e) {
+            resultSets[0] = ProcedureUtils.generateResult("Error", e.getLocalizedMessage());
```

Comment:
is there any special reason to modify a method parameter and to use it as a result? It would be cleaner to generate an object inside of the method. Another issue if the resultSets array will come empty it will throw an exception.

Reply:
This follows our system procedure framework. The result set is returned as OUT parameter and passed to CallStatementOperation.call()